### PR TITLE
feat: retire Beck agent — fold testing into SHARED.md

### LINKED ISSUE #644: Retire Beck agent — fold testing guidelines into SHARED.md
## Context

Beck (test implementer) has never been spawned as an implementing agent across the entire project history. Every test has been written by the implementing agent (Voss, Deming, Hamilton) inline with their work. The Knuth → Beck pipeline (audit findings → test cases) has never been exercised — Knuth findings go directly to the implementing agent.

Beck adds a routing hop that provides no value and a 15th agent definition to maintain.

## Plan

1. **Add "When writing tests" section to `templates/agents/SHARED.md`** with Beck's key guidelines:
   - Test isolation: no shared state, no execution order dependencies
   - Meaningful assertions: no false-confidence tests
   - Anti-mocking: push back on over-mocking, testability is a design quality
   - Test naming as documentation: names describe behavior, not implementation
   - Fixture/teardown: minimal setup, reliable teardown, immutable shared fixtures

2. **Remove Beck from the task skill routing table** in `templates/skills/dev-team-task/SKILL.md` — "Test writing" routes to the implementing agent for the area under test (Voss for backend, Deming for tooling, etc.)

3. **Remove Beck agent definition** from `templates/agents/dev-team-beck.md`

4. **Remove Beck from agent lists** in `templates/CLAUDE.md`, `CLAUDE.md`, config references

5. **Remove Beck agent memory** from `templates/agent-memory/dev-team-beck/`

6. **Update `src/init.ts`** — remove Beck from the agents array if hardcoded

7. **Add migration** in `src/update.ts` to clean up Beck files in existing installations

## Acceptance criteria

- [ ] `SHARED.md` has a "When writing tests" section with Beck's guidelines
- [ ] Task skill routing table no longer references Beck
- [ ] Beck agent definition, memory template removed
- [ ] All CLAUDE.md agent lists updated
- [ ] Migration removes Beck from existing installations
- [ ] `npm test` passes

### PR DESCRIPTION
**Pull request by fredericboyer**
## Summary

- Folds Beck's testing guidelines (isolation, assertions, anti-mocking, naming, fixtures) into `SHARED.md` so all agents inherit them
- Removes Beck agent definition, memory template, and all references across 31 files
- Adds `agentRemovals` migration support in `update.ts` to clean up Beck files in existing installations
- Updates tests: removes Beck-specific tests, adjusts agent counts (14->13, 15->14)

Closes #644

## Test plan

- [x] `npm run build` passes
- [x] `npm test` — 614 tests, 0 failures
- [x] `node scripts/validate-agents.js` — 13 agents valid
- [x] No Beck references remain in templates, src, hooks, or skills
- [ ] CI passes

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -124,7 +123,6 @@ npx @fredericboyer/dev-team create-agent <name>     # Scaffold a custom agent
 | `@dev-team-hamilton` | Infrastructure Engineer | sonnet | Dockerfiles, IaC, CI/CD, k8s, deployment, monitoring |
 | `@dev-team-szabo` | Security Auditor | opus | Vulnerability review, auth flows, attack surfaces |
 | `@dev-team-knuth` | Quality Auditor | opus | Coverage gaps, boundary conditions, correctness |
```
[the remark this answers is not in the document]
Comment (fredericboyer) [reply README.md]: Tracked in #644 — integration test for agentRemovals migration is a valid follow-up

File: src/update.ts [hunk 2]
```diff
@@ -70,6 +76,10 @@ const MIGRATIONS: Migration[] = [
     version: "1.0.0",
     skillRemovals: ["dev-team-merge", "dev-team-security-status"],
   },
+  {
+    version: "3.2.0",
```
[the remark this answers is not in the document]
Comment (fredericboyer) [reply src/update.ts]: Fixed: Changed migration version from 3.2.0 to 3.1.1
